model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/changes/UserSessionPersistentChangelogBasedTransaction.java

```diff
@@ -57,6 +57,7 @@ public SessionEntityWrapper<UserSessionEntity> get(RealmModel realm, String key,
                 LOG.debugf("user-session not found in cache for sessionId=%s offline=%s, loading from persister", key, offline);
                 wrappedEntity = getSessionEntityFromPersister(realm, key, userSession, offline);
             } else {
+                getUpdates(offline).putIfAbsent(key, new SessionUpdatesList<>(realm, wrappedEntity));
```

Comment:
I fail to understand why this was moved. Shouldn't we add the wrapped entity to the updates list?

Reply:
As discussed, the `getSessionEntityFromPersister` already adds it to the map. Also this change to `putIfAbsent` prevents overwriting existing entries